### PR TITLE
Refactor requestability to delegate to the item

### PR DESCRIPTION
**Pull request by thatbudakguy**
This PR fixes a couple of inconsistencies (typos, issues with fixtures) and does one main refactor: making `Folio::Item` the source of truth for requestability via various methods. This means that:
- `RequestAbilities` can use the items in the request to figure out whether the overall request is viable
- `SubmitFolioRequestJob` can use the items in the request to figure out how to submit the job
- The item selector view can use the items to figure out which ones should be requestable/checkable

As a result, the tests for `SubmitFolioRequestJob` were significantly expanded and check for more things (like whether the request is actually the right type, and who it is for).

This came out of/is pre-work for #1776, because we will need to do things like enable/disable request checkboxes and set the status text/icons on a per-item basis using this logic.

### CODE REVIEW

File: app/models/folio/item.rb
Comment (jcoyne) [inline app/models/folio/item.rb:142]: allowed_request_types could be nil, so you can't send `include?` to it without checking for nil.

File: app/models/folio/item.rb
Comment (jcoyne) [inline app/models/folio/item.rb:146]: allowed_request_types could be nil, so you can't send `include?` to it without checking for nil.

File: app/models/folio/item.rb
Comment (jcoyne) [inline app/models/folio/item.rb:150]: allowed_request_types could be nil, so you can't send `include?` to it without checking for nil.

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: app/models/folio/item.rb
Comment (thatbudakguy) [reply app/models/folio/item.rb:142]: Added a check to make it `[]` if nil

**Review by jcoyne — APPROVED** [review]

**Review by cbeer — CHANGES_REQUESTED** [review]
I think (someone) should take this for a spin locally to do some UAT before we merge this. (Using Request changes to block merge for now). Some scenarios:

- [ ] On-campus paging (e.g. from BUSINESS)
- [x] Location-restricted paging (e.g. SAL3-PAGE-SP)
- [x] Hold/recall from a non-pageable location
- [ ] Hold/recall from a pageable location
- [ ] Hold/recall for an in-process item
- [ ] Title-level hold

**Comment by cbeer** [comment]
I'm seeing a new stack trace paging a non-existing item:

> undefined method `items' for nil:NilClass

(Coming from Request#paging_origin_library).

I don't think we need to handle that scenario gracefully, but maybe we can do better than a NoMethodError.

**Comment by cbeer** [comment]
I'm also getting a `PagesController::UnpageableItemError` error paging a checked out item from BUS-SAL3-STACKS (e.g. https://searchworks-folio-test.stanford.edu/view/10000041). It might be a circ rule thing, but I think this should either be a sent down the hold/recall path or use the paging form (but submit a hold-recall).

**Comment by cbeer** [comment]
https://searchworks.stanford.edu/view/13956859 is now in a weird state too; maybe the problem is we're showing a request button, but it also gives a `PagesController::UnpageableItemError` (although I thought Missing was perhaps hold/recall-able?)

**Comment by thatbudakguy** [comment]
> I'm seeing a new stack trace paging a non-existing item:

I added a pathway where this raises a 404.

> I'm also getting a PagesController::UnpageableItemError error paging a checked out item from BUS-SAL3-STACKS (e.g. https://searchworks-folio-test.stanford.edu/view/10000041).

This thing is `Aged to lost`, which FOLIO [doesn't let you request at all](https://github.com/folio-org/mod-circulation/blob/master/src/main/java/org/folio/circulation/domain/RequestTypeItemStatusWhiteList.java#L71-L94). For now, it seems like we need to suppress the request button in SearchWorks.

> https://searchworks.stanford.edu/view/13956859 is now in a weird state too; maybe the problem is we're showing a request button, but it also gives a PagesController::UnpageableItemError (although I thought Missing was perhaps hold/recall-able?)

Added a check so that in this case (no barcode sent, one holdable but not pageable item, two non-requestable items) we fall back to assuming the request is a Page and let the items figure it out after the user selects one.

**Review by cbeer — APPROVED** [review]

**Pull request merged**
